tests/TransactionBuilderTest.php

```diff
@@ -0,0 +1,220 @@
+        DB::beginTransaction();
+        DB::collection('users')->where('name', $this->originData['name'])->update(['age' => $new_age]);
+        DB::rollBack();
+        $users = DB::collection('users')->where('name', $this->originData['name'])->where('age', $new_age)->get();
+        $this->assertCount(0, $users);
+
+        /** commit test */
+        DB::beginTransaction();
+        DB::collection('users')->where('name', $this->originData['name'])->update(['age' => $new_age]);
+        DB::commit();
+        $users = DB::collection('users')->where('name', $this->originData['name'])->where('age', $new_age)->get();
+        $this->assertCount(1, $users);
+    }
+
+    public function testDelete()
+    {
+        /** rollback test */
+        DB::beginTransaction();
+        DB::collection('users')->where('name', $this->originData['name'])->delete();
+        DB::rollBack();
+        $users = DB::collection('users')->where('name', $this->originData['name'])->get();
+        $this->assertCount(1, $users);
+
+        /** commit test */
+        DB::beginTransaction();
+        DB::collection('users')->where('name', $this->originData['name'])->delete();
+        DB::commit();
+        $users = DB::collection('users')->where('name', $this->originData['name'])->get();
+        $this->assertCount(0, $users);
+    }
+
+    public function testIncrement()
+    {
+        /** rollback test */
+        DB::beginTransaction();
+        DB::collection('users')->where('name', $this->originData['name'])->increment('age');
+        DB::rollBack();
+        $user = DB::collection('users')->where('name', $this->originData['name'])->first();
+        $this->assertEquals($this->originData['age'], $user['age']);
+
+        /** commit test */
+        DB::beginTransaction();
+        DB::collection('users')->where('name', $this->originData['name'])->increment('age');
+        DB::commit();
+        $user = DB::collection('users')->where('name', $this->originData['name'])->first();
+        $this->assertEquals($this->originData['age'] + 1, $user['age']);
+    }
+
+    public function testDecrement()
+    {
+        /** rollback test */
+        DB::beginTransaction();
+        DB::collection('users')->where('name', $this->originData['name'])->decrement('age');
+        DB::rollBack();
+        $user = DB::collection('users')->where('name', $this->originData['name'])->first();
+        $this->assertEquals($this->originData['age'], $user['age']);
+
+        /** commit test */
+        DB::beginTransaction();
+        DB::collection('users')->where('name', $this->originData['name'])->decrement('age');
+        DB::commit();
+        $user = DB::collection('users')->where('name', $this->originData['name'])->first();
+        $this->assertEquals($this->originData['age'] - 1, $user['age']);
+    }
+
+    public function testQuery()
+    {
+        /** rollback test */
+        DB::beginTransaction();
+        $count = DB::collection('users')->count();
```

Comment:
Does this operation use the `count` command or `aggregate`? I'll note that `count` is not supported within transactions (see: [spec](https://github.com/mongodb/specifications/blob/master/source/transactions/transactions.rst#what-commands-can-be-run-in-a-transaction)).

Reply:
Under the hood `count` call the `aggregate`. 
```
/**
* Retrieve the "count" result of the query.
*
* @param  string  $columns
* @return int
*/
public function count($columns = '*')
{
   return (int) $this->aggregate(__FUNCTION__, Arr::wrap($columns));
}
```